tests/test_web/test_api_v1.py

```diff
@@ -521,6 +533,30 @@ def test_add_bundles_overwrite_not_allowed(mock_smfsc, client, db):
     mock_smfsc.assert_not_called()
 
 
+@pytest.mark.parametrize('from_index', (None, 'some-common-index'))
+@mock.patch('iib.web.api_v1.messaging.send_message_for_state_change')
+def test_add_bundles_graph_update_mode_not_allowed(
+    mock_smfsc, app, client, auth_env, db, from_index
+):
+    app.config['IIB_GRAPH_MODE_INDEX_ALLOW_LIST'] = ['some-unique-index']
+    data = {
+        'binary_image': 'binary:image',
+        'bundles': ['some:thing'],
+        'from_index': from_index,
```

Comment:
Why are we using @pytest.mark.parametrize for from_index here? Could it be set, as other values in the 'data'?

Reply:
yes, so it basically creates two separate tests for each value. I parameterized it just to make sure that IIB will fail when no `from_index` is specified or a pullspec not present in the config option is specified.

Reply:
I see now, I misunderstood the functionality of that decorator.